File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        
        
        pivot =0
        
        for char in t:
            
            if pivot==len(s):
                return True
            
            elif s[pivot]==char:
                pivot+=1
                
        return pivot ==len(s)

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        
        int pivot = 0;
        
        for(int i=0; i<t.length(); i++ ){
            
            if(pivot == s.length()) {
                return true;
            } else if(s.charAt(pivot)==t.charAt(i)) {
                pivot++;
            }
            
        }
        
        return s.length() ==pivot;
        
    }
}